.github/workflows/run-helm.yml

```diff
@@ -27,6 +27,33 @@ env:
   helm_version: "3.2.1"
 
 jobs:
+  lint-with-current-helm:
+    # Run workflow pull_request if it is NOT a fork, as pull_request_target if it IS a fork
+    if: >-
+      ( github.event_name == 'pull_request_target' && github.event.pull_request.head.repo.fork == true ) ||
+      ( github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == false )
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - name: Checkout the merged commit from PR and base branch
+        uses: actions/checkout@v2
+        if: github.event_name == 'pull_request_target'
+        with:
+          # pull_request_target is run in the context of the base repository
+          # of the pull request, so the default ref is master branch and
+          # ref should be manually set to the head of the PR
+          ref: refs/pull/${{ github.event.pull_request.number }}/head
+
+      - name: Checkout the head commit of the branch
+        if: ${{ github.event_name != 'pull_request_target' }}
+        uses: actions/checkout@v2
+
+      - uses: azure/setup-helm@v1
+
+      - name: Lint helm chart
+        run: helm lint deployment/helm && echo "lint finished successfully" || echo ::warning::"lint found issues"
```

Comment:
Would the echo hide the failure (overall exit code will be 0); or do you not want this step to cause failure of the pipeline? Wondering if it is not a failure, if it will not get noticed.

Reply:
Thinking more, if we are hardcoding the helm version, should we not leave it as failure? Warning would make more sense if we were on the latest. (or should we stop hardcoding now that it would just be a warning?)

Reply:
there are two parts to this:
1. the hard-coded version: the linting here will cause a failure
1. the non-specified, most recent version: the linting here will not cause a failure and we will have to look to find the warning (it will not show up anywhere)

my intent is to not allow random version changes to Helm cause a failure in our CI.

but to also allow us to see any potential issues with upcoming versions of Helm.  there is no way to bubble up a "warning" without causing a failure in the CI.  so this is my best approximation.